sources/web/datalab/jupyter.ts

```diff
@@ -342,9 +343,29 @@ export function handleRequest(request: http.ServerRequest, response: http.Server
     response.end();
     return;
   }
+
+  var path = url.parse(request.url).pathname;
+  if (path.indexOf('/sessions') == 0) {
```

Comment:
> Can we move '/tree' and '/edit' here too?

Not without consequences.

Specifically, we rely on the Jupyter server validating the path passed to each of the '/tree', '/edit', and '/nb' handlers. As such, if we changed those handlers to be statically served by the Node JS server, then instead of a 404 message, the client would get the regular HTML page with API calls that will fail.